arch/arc/core/mpu/arc_mpu.c

```diff
@@ -417,6 +438,37 @@ void arc_core_mpu_region(u32_t index, u32_t base, u32_t size,
 }
 
 #if defined(CONFIG_USERSPACE)
+void arc_core_mpu_configure_user_context(struct k_thread *thread)
+{
+	u32_t base = (u32_t)thread->stack_obj;
+	u32_t size = thread->stack_info.size;
+
+	/* for kernel threads, no need to configure user context */
+	if (!thread->arch.priv_stack_start) {
+		return;
+	}
+
+	arc_core_mpu_configure(THREAD_STACK_USER_REGION, base, size);
+
+	/* configure app data portion */
+#ifdef CONFIG_APPLICATION_MEMORY
+#if CONFIG_ARC_MPU_VER == 2
+	base = (u32_t)&__app_ram_start;
+	size = (u32_t)&__app_ram_end - (u32_t)&__app_ram_start;
```

Comment:
what happens if size is not power-of-two aligned?

Reply:
the arm mpu code rounded up sizes to power of two.  I'd expect the ARC to do something similar to choose the right size value.  As an aside, the section alignment needs to be taken into consideration as it needs to be aligned to the rounded up value and padded out

Reply:
In my latest commits, i already apply the python script (gen_alignment_script.py) to generate the correct address alignment for application memory for power-of-2 MPU, i.e., ARC MPUv2. For ARC MPUv3, necessary definitions are added in linker template. 

I will also add some checks or processing here to guarantee the correct start address and size